### PR TITLE
Support atropisomers in the conformer generator

### PR DESCRIPTION
**Pull request by greglandrum**
The existing logic for handling atomic chirality could be extended to work for atropisomers which have two exo substituents on at least one side.

Supporting a single exo substituent on both sides is likely to be trickier, but given that the barriers in those systems are likely to be quite low, it seems less relevant. Still, it's something we can look at later.

### CODE REVIEW

File: Code/GraphMol/DistGeomHelpers/Embedder.cpp [hunk 1]
```diff
@@ -780,6 +792,8 @@ bool finalChiralChecks(RDGeom::PointPtrVect *positions,
       return false;
     }
   }
+  // FIX: do we need some kind of sanity check here for the non-atomic
+  // situations (e.g. atropisomers)?
```
[the remark this answers is not in the document]
Comment (greglandrum) [reply Code/GraphMol/DistGeomHelpers/Embedder.cpp:796]: We have some extra checks for normal chiral centers that look at whether nor not the local environment around the atom looks ok. These aren't great, but we added them in response to bad conformers that we observed.
The idea of this comment was to flag the place similar checks would go if we start seeing fishy-looking atropisomer conformers

[the remark this answers is not in the document]
Comment (bp-kelley) [reply Code/GraphMol/DistGeomHelpers/Embedder.cpp:796]: I have a handful of smarts patterns that we use for atropisomer detection/possibility which potentially can be the basis of sanity checks.

[the remark this answers is not in the document]
Comment (greglandrum) [reply Code/GraphMol/DistGeomHelpers/Embedder.cpp:796]: > I have a handful of smarts patterns that we use for atropisomer detection/possibility which potentially can be the basis of sanity checks.

I think we're talking about different things. The idea here would be to check to see if a generated conformer actually matches what the chirality tag says it should be.

I think the SMARTS would be more about detecting whether or not a particular bond should be considered atropisomeric. That's obviously something people can do on their own as part of their preprocessing pipeline, but we decided against doing it at the toolkit level. If there's wedging around a bond which might have atropisomerism, then we consider it atropisomeric. This (and what types of bonds are eligible) is documented in the RDKit book.

**Comment by bp-kelley** [comment]
I should finish the review this week.

**Review by ptosco — APPROVED** [review]
To the extent that I understand how the code works, it looks good to me.

**Pull request merged**